packages/ai-jsx/src/batteries/sidekick/platform/conversation.tsx

```diff
@@ -81,15 +94,39 @@ export function getNextConversationStep(
 ) {
   const shrinkableConversation = getShrinkableConversation(messages, fullConversation);
   const lastMessage = messages[messages.length - 1];
+
+  // Add tools for interacting with redacted function responses (if one exists).
+  // Will only take into account the current round of messages. After that the LLM
+  // will need to call the function again. This is to prevent the LLM from accessing stale data.
+  const updatedTools = { ...tools, ...redactedFunctionTools(getCurrentRound(fullConversation)) };
+
   switch (lastMessage.type) {
     case 'functionCall': {
       const { name, args } = lastMessage.element.props;
-      return <ExecuteFunction func={tools[name].func} name={name} args={args} />;
+      const executedFunction = (
+        <ExecuteFunction
+          // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
```

Comment:
Are the types wrong here?

Reply:
I'm not sure why TypeScript doesn't throw an error here. There is no reason why `name` should be available in `tools`.
I also noticed that @petersalas had handled the case where `func` was `undefined`, but from what I understand, this line will error out before getting to that point since `tool[name]` can be the thing that is `undefined` (usually if the tool exists, the `func` also exists, but no guarantee that the `tool` exists).